kernels/test/op_convolution_backward_test.cpp

```diff
@@ -62,139 +62,155 @@ class OpConvolutionBackwardOutTest : public OperatorTest {
+        40243, 25890, 40472, 30460, 21228, 21625, 13289, 24435, 19876, 29816,
+        24188, 23619, 13752, 16251, 18741, 19368, 24517, 34261, 27054, 31257,
+        21238, 18909, 15776, 16881, 34604, 22534, 28101, 23834, 18479, 16469,
+        12852, 16551, 14204, 29983, 20167, 24150, 14281, 17501, 15897, 16019,
+        21661, 32765, 23874, 26527, 20463, 18661};
+    std::vector<CTYPE> expected_grad_bias_data = {363, 438, 585, 501};
 
-  auto grad_output = tf.make({2, 4, 3, 2}, grad_output_data);
-  auto input = tf.make({2, 6, 7, 5}, input_data);
-  auto weight = tf.make({4, 3, 4, 2}, weight_data);
-  int64_t bias_sizes[1] = {4};
-  int64_t stride[2] = {1, 2};
-  int64_t padding[2] = {1, 0};
-  int64_t dilation[2] = {2, 1};
-  bool transposed = false;
-  int64_t output_padding[2] = {0, 0};
-  int64_t groups = 2;
-  std::array<bool, 3> output_mask_a = {true, true, true};
-  auto grad_input = tf.zeros({2, 6, 7, 5});
-  auto grad_weight = tf.zeros({4, 3, 4, 2});
-  auto grad_bias = tf.zeros({4});
+    auto grad_output = tf.make({2, 4, 3, 2}, grad_output_data);
+    auto input = tf.make({2, 6, 7, 5}, input_data);
+    auto weight = tf.make({4, 3, 4, 2}, weight_data);
+    int64_t bias_sizes[1] = {4};
+    int64_t stride[2] = {1, 2};
+    int64_t padding[2] = {1, 0};
+    int64_t dilation[2] = {2, 1};
+    bool transposed = false;
+    int64_t output_padding[2] = {0, 0};
+    int64_t groups = 2;
+    std::array<bool, 3> output_mask_a = {true, true, true};
+    auto grad_input = tf.zeros({2, 6, 7, 5});
+    auto grad_weight = tf.zeros({4, 3, 4, 2});
+    auto grad_bias = tf.zeros({4});
+
+    op_convolution_backward_out(
+        grad_output,
+        input,
+        weight,
+        IntArrayRef{bias_sizes, 1},
+        IntArrayRef{stride, 2},
+        IntArrayRef{padding, 2},
+        IntArrayRef{dilation, 2},
+        transposed,
+        IntArrayRef{output_padding, 2},
+        groups,
+        output_mask_a,
+        grad_input,
+        grad_weight,
+        grad_bias);
 
-  op_convolution_backward_out(
-      grad_output,
-      input,
-      weight,
-      IntArrayRef{bias_sizes, 1},
-      IntArrayRef{stride, 2},
-      IntArrayRef{padding, 2},
-      IntArrayRef{dilation, 2},
-      transposed,
-      IntArrayRef{output_padding, 2},
-      groups,
-      output_mask_a,
-      grad_input,
-      grad_weight,
-      grad_bias);
+    auto expected_grad_input = tf.make({2, 6, 7, 5}, expected_grad_input_data);
+    auto expected_grad_weight =
+        tf.make({4, 3, 4, 2}, expected_grad_weight_data);
+    auto expected_grad_bias = tf.make({4}, expected_grad_bias_data);
 
-  auto expected_grad_input = tf.make({2, 6, 7, 5}, expected_grad_input_data);
-  auto expected_grad_weight = tf.make({4, 3, 4, 2}, expected_grad_weight_data);
-  auto expected_grad_bias = tf.make({4}, expected_grad_bias_data);
+    if (DTYPE == ScalarType::Half || DTYPE == ScalarType::BFloat16) {
+      EXPECT_TENSOR_CLOSE_WITH_TOL(grad_input, expected_grad_input, 1e-2, 1e-8);
```

Comment:
Why not use defaults here? `EXPECT_TENSOR_CLOSE_WITH_TOL` should apply the right tolerance given the type

Reply:
because the default rtol is 1e-5; rtol and atol are different

Reply:
right, but in the same way that we have `kDefaultHalfAtol` and `kDefaultBFloat16Atol` I think we should have `kDefaultHalfRtol` and `kDefaultBFloat16Rtol` and set it to a proper value.
You seem to be using 1e-2 for most of these tests. Why not introduced `kDefaultHalfRtol` and `kDefaultBFloat16Rtol` with value 1e-2?

Reply:
> Why not introduced kDefaultHalfRtol and kDefaultBFloat16Rtol with value 1e-2?

Because not all operators require the higher rtol.

Reply:
group_norm is one of the ops that automatic mixed precision will autocast to float32: https://intel.github.io/intel-extension-for-pytorch/xpu/1.10.200+gpu/tutorials/features/amp.html

I think the norm ops are just particularly prone to roundoff error, but I'm certainly not a numerical analysis person.